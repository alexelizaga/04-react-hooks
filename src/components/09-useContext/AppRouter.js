import React from 'react';
import {
  BrowserRouter,
  Routes,
  Route,
  Navigate
} from 'react-router-dom';

import { NavBar } from './components/NavBar';
import { HomeScreen } from './screens/HomeScreen';
import { LoginScreen } from './screens/LoginScreen';
import { AboutScreen } from './screens/AboutScreen';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <div>
        <NavBar />
        <Routes>
          <Route exact path='/' element={ <HomeScreen /> } />
          <Route path='/login' element={ <LoginScreen /> } />
          <Route path='/about' element={ <AboutScreen /> } />

          <Route path='*'  element={<Navigate to="/" replace />} />
        </Routes>
      </div>
    </BrowserRouter>
  )
}
