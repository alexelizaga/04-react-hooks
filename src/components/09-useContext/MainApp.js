import React from 'react'

import { AppRouter } from './AppRouter';
import { UserContext } from './context/UserContext';

export const MainApp = () => {
  const user = {
    id: 1234,
    name: 'Alex',
    email: 'alex@alex.com'
  }
  return (
    <UserContext.Provider value={user}>
      <AppRouter/>
    </UserContext.Provider>
  )
}
