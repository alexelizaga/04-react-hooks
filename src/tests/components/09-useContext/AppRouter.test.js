import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';

import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/context/UserContext';

describe('Test <AppRouter />', () => {

  const user = {
    id: 1,
    name: 'Alex'
  }

  test('should render correctly', () => {
    const tree = renderer.create(
      <UserContext.Provider value={{
        user
      }}>
        <AppRouter />
      </UserContext.Provider>
      
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});