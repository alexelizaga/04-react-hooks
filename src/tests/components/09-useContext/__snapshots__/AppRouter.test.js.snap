// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AppRouter /> should render correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      useContext
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
    "id": 1,
    "name": "Alex"
}
      </pre>
    </div>
  </div>
</div>
`;
